src/prommis/uky/costing/ree_plant_capcost.py

```diff
@@ -296,648 +454,673 @@ def build_process_costs(
-                    self.Lang_factor - 1
+                self.total_installation_cost_eq = Constraint(
+                    rule=total_installation_cost_rule
                 )
 
-            self.total_installation_cost_eq = Constraint(
-                rule=total_installation_cost_rule
-            )
-
-        # rule for calculating TPC
-        def total_plant_cost_rule(self):
-            return self.total_plant_cost == (
-                self.total_BEC + self.total_installation_cost + self.other_plant_costs
-            )
+            # rule for calculating TPC
+            def total_plant_cost_rule(self):
+                return self.total_plant_cost == (
+                    self.total_BEC
+                    + self.total_installation_cost
+                    + self.other_plant_costs
+                )
 
-        self.total_plant_cost_eq = Constraint(rule=total_plant_cost_rule)
+            self.total_plant_cost_eq = Constraint(rule=total_plant_cost_rule)
 
-        # define land cost
-        if land_cost is not None:
-            if type(land_cost) in [Expression, ScalarExpression]:
-                if pyunits.get_units(land_cost) == pyunits.dimensionless:
-                    self.land_cost = Expression(expr=land_cost.expr * CE_index_units)
+            # define land cost
+            if land_cost is not None:
+                if type(land_cost) in [Expression, ScalarExpression]:
+                    if pyunits.get_units(land_cost) == pyunits.dimensionless:
+                        self.land_cost = Expression(
+                            expr=land_cost.expr * CE_index_units
+                        )
+                    else:
+                        self.land_cost = Expression(
+                            expr=pyunits.convert(
+                                land_cost.expr, to_units=CE_index_units
+                            )
+                        )
                 else:
-                    self.land_cost = Expression(
-                        expr=pyunits.convert(land_cost.expr, to_units=CE_index_units)
-                    )
+                    if pyunits.get_units(land_cost) == pyunits.dimensionless:
+                        self.land_cost = Expression(expr=land_cost * CE_index_units)
+                    else:
+                        self.land_cost = Expression(
+                            expr=pyunits.convert(land_cost, to_units=CE_index_units)
+                        )
             else:
-                if pyunits.get_units(land_cost) == pyunits.dimensionless:
-                    self.land_cost = Expression(expr=land_cost * CE_index_units)
+                self.land_cost = Expression(expr=0 * CE_index_units)
+
+            # define feed input, if passed
+            if feed_input is not None:
+                if (
+                    pyunits.get_units(feed_input) == pyunits.dimensionless
+                ):  # assume it's short ton per hour
```

Comment:
Is this a good assumption? Generally when there is any ambiguity (or in this case an outright unit inconsistency) I generally recommend raising an Exception instead of making an assumption (which is quite likely wrong).